src/github.com/matrix-org/dendrite/syncserver/storage/output_room_events_table.go

```diff
@@ -244,7 +267,14 @@ func rowsToEvents(rows *sql.Rows) ([]gomatrixserverlib.Event, error) {
 		if err != nil {
 			return nil, err
 		}
-		result = append(result, ev)
+		if order == rowOrderForward {
+			result = append(result, ev)
+		} else if order == rowOrderReverse {
+			result = append([]gomatrixserverlib.Event{ev}, result...)
```

Comment:
Sorry I guess it's technically O(n) for the appending to the end each time. I think sum of 2^k for k from 0 to log(n) is n?